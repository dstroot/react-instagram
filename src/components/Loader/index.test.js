import React from 'react';
import { render, cleanup } from '@testing-library/react';

// component to test
import { Loader } from '.';

// should be in setupTests.js?
afterEach(cleanup);

// The test calls the render function provided by React Testing Library,
// which returns an object that we can extract asFragment from. Using
// this we can use the built-in functionality from Jest to perform a snapshot test.
it('renders', () => {
  const { asFragment } = render(
    <Loader color="#1089ff" height={100} width={100} />
  );
  expect(asFragment()).toMatchSnapshot();
});
