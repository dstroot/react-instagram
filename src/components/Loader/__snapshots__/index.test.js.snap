// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <svg
    aria-label="loading"
    height="100"
    stroke="#1089ff"
    viewBox="0 0 38 38"
    width="100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fill="none"
      fill-rule="evenodd"
    >
      <g
        stroke-width="2"
        transform="translate(1 1)"
      >
        <circle
          cx="18"
          cy="18"
          r="18"
          stroke-opacity=".5"
        />
        <path
          d="M36 18c0-9.94-8.06-18-18-18"
        >
          <animatetransform
            attributeName="transform"
            dur="1s"
            from="0 18 18"
            repeatCount="indefinite"
            to="360 18 18"
            type="rotate"
          />
        </path>
      </g>
    </g>
  </svg>
</DocumentFragment>
`;
